saas/models/saas_template.py

```diff
@@ -221,21 +222,22 @@ def create_db(self, key_values=None, db_name=None, with_delay=True, draft_build_
 
         self.env['saas.log'].log_db_creating(build, self.operator_db_id)
         if with_delay:
-            build.with_delay().create_db(
+            job_create=(build.with_delay().create_db(
                 self.operator_db_name,
                 self.template_id.template_demo,
-            )
-            build.with_delay().action_install_missing_mandatory_modules()
+            ))
+            job_missing_mandatory=(group(job_create).on_done(
+            build.with_delay().action_install_missing_mandatory_modules()))
 
-            self.operator_id.with_delay().build_post_init(build, self.template_id.build_post_init, key_values)
+            group(job_missing_mandatory).on_done(self.operator_id.with_delay().build_post_init(build, self.template_id.build_post_init, key_values))
         else:
-            build.create_db(
+            job_create=(build.create_db(
                 self.operator_db_name,
                 self.template_id.template_demo,
-            )
-            build.action_install_missing_mandatory_modules()
+            ))
+            job_missing_mandatory=(group(job_create).on_done(build.action_install_missing_mandatory_modules()))
 
-            self.operator_id.build_post_init(build, self.template_id.build_post_init, key_values)
+            group(job_missing_mandatory).on_done(self.operator_id.build_post_init(build, self.template_id.build_post_init, key_values))
```

Comment:
Why do we need to add jobs here, since all actions here are expected to run synchronically in one thread?

Reply:
Hello,

I wanted to keep the same architecture of the function, so my goal was not to rethink the operation but rather to accentuate the order of executions.
I didn't want to generate bugs, hence the fact that I did separate the executions.

Reply:
Hello,

I wanted to keep the same architecture of the function, so my goal was not to rethink the operation but rather to accentuate the order of executions.
I didn't want to generate bugs, hence the fact that I did separate the executions.

Reply:
I removed this part in commit mentioned below. If we run this method (saas.template.create_db) with with_delay=False, it is expected that database will created, modules installed and post init done after this method is executed. So there should not be any delayed job here.

Also issue https://github.com/it-projects-llc/saas-addons/issues/134 is not reproduced if we run this method using with_delay = False argument.